servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/TransportObservers.java

```diff
@@ -36,6 +38,31 @@ public static TransportObserver asSafeObserver(final TransportObserver observer)
         if (observer instanceof CatchAllTransportObserver) {
             return observer;
         }
+        if (observer instanceof BiTransportObserver) {
+            // BiTransportObserver is always safe
+            return observer;
+        }
         return new CatchAllTransportObserver(observer);
     }
+
+    /**
+     * Combines multiple {@link TransportObserver}s into a single {@link TransportObserver}.
+     *
+     * @param other {@link TransportObserver}s to combine
+     * @return a {@link TransportObserver} that delegates all invocations to the provided {@link TransportObserver}s
+     */
+    public static TransportObserver combine(final TransportObserver... other) {
+        switch (other.length) {
+            case 0:
+                throw new IllegalArgumentException("At least one TransportObserver is required to combine");
```

Comment:
This is what I was trying to avoid by having `first, second, others` arguments: http://jtechies.blogspot.com/2012/07/item-42-use-varargs-judiciously.html

Reply:
I think this is OK and inline with what we do elsewhere:

https://github.com/apple/servicetalk/blob/main/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/MergeCompletable.java#L41